src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py

```diff
@@ -376,14 +376,17 @@ def evaluate(
             columns=[col for col in evaluator_result_df.columns if col.startswith("inputs.")]
         )
 
-        # rename output columns
-        # Assuming after removing inputs columns, all columns are output columns
-        evaluator_result_df.rename(
-            columns={
-                col: "outputs." f"{evaluator_name}.{col.replace('outputs.', '')}" for col in evaluator_result_df.columns
-            },
-            inplace=True,
-        )
+        # If we use thread pool, the columns are already renamed.
+        if not use_thread_pool:

```

Comment:
I don't see the code change for the error or empty value handling for some reason.

Reply:
We are already handling the empty values [here](https://github.com/microsoft/promptflow/blob/235cb01a43e8d354909a04349cca3421bca97a1c/src/promptflow-evals/promptflow/evals/evaluate/_code_client/code_client.py#L42). If the evaluator breaks, we just add the empty line to output data frame. Should we change this behavior?